tools/compress_assets.py

```
@@ -0,0 +1,141 @@
+"""
+Pre-build script for compressing web assets (CSS, JS) into gzipped binary arrays.
+This script runs before each build and generates src/web_assets.h
+"""
+
+import os
+import gzip
+import hashlib
+
+# Check if running within PlatformIO build environment
+try:
+    Import("env")
```

Ruff (0.15.2)

[error] Undefined name `Import`

(F821)